Core/PlayerPersistentData.gd

```diff
@@ -0,0 +1,7 @@
+extends Resource
+class_name PlayerPersistentData
+## Store player data that we want to persist between scenes here.
+
+
+var saved_health: float = -1
+var saved_stats: EntityStats = null
```

Comment:
I would use an `_init` to set the default values
i had some problem later in other parts of the code where if you extend class from this class and you want different default value, you won't be able to do it if you don't use an `_init`. Probably not useful here, but for the sake of consistency